### PR TITLE
Allow pin drivers as wakeup sources

### LINKED ISSUE #2020: Used GPIOs can't be unconfigured for wakeup
If I move a GPIO to be used for something, I can't get the struct back, so I can't (using only safe code) configure the GPIO to be a wakeup source.

### PR DESCRIPTION
**Pull request by bugadani**
Closes #2020

### CODE REVIEW

File: esp-hal/src/gpio/lp_io.rs [hunk 1]
```diff
@@ -257,6 +257,24 @@ macro_rules! lp_gpio {
                     }
                 }
             )+
+
+            #[doc(hidden)]
+            #[macro_export]
+            macro_rules! handle_rtcio {
```
Comment (bugadani) [inline esp-hal/src/gpio/lp_io.rs:263]: Next up, how can I delete these... I'm hoping there is a solution that allows me to factor this as a `repeat_erased!` and a `filter_branches!` macro, but I haven't tried anything yet.

File: esp-hal/src/gpio/mod.rs [hunk 2]
```diff
@@ -2583,17 +2489,82 @@ pub(crate) mod internal {
...
+            handle_gpio_output!(&self.0, target, {
                 OutputPin::is_set_high(target, private::Internal)
             })
         }
     }
+
+    #[cfg(any(xtensa, esp32c2, esp32c3, esp32c6))]
+    impl RtcPin for ErasedPin {
+        #[cfg(xtensa)]
+        #[allow(unused_braces)] // False positive :/
```
Comment (bugadani) [inline esp-hal/src/gpio/mod.rs:2507]: https://github.com/rust-lang/rust/issues/88104

**Review by bjoernQ — APPROVED** [review]
LGTM - makes things more convenient!

**Review by MabezDev — APPROVED** [review]
Thanks!

**Pull request merged**